### PR TITLE
Fix upgrade (don't keep deleted files from older upstream versions)

### PR DESCRIPTION
**Pull request by JimboJoe**
## Problem
- *Upgrade to 19.2 introduced UI problems due to remaining files from previous upstream version*

## Solution
- *Keep only configuration file when upgrading*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : Maniack C
- [x] **Approval (LGTM)** : Maniack C
- [x] **Approval (LGTM)** : Kay0u
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/ttrss_ynh%20fix_upgrade%20(Official)/)
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by JimboJoe** [comment]
Obviously current testing #59 should wait for this fix to be released as stable.

**Comment by maniackcrudelis** [comment]
Looks good, trust you know what you're doing.

**Comment by JimboJoe** [comment]
You can trust that it fixed my production server :sweat_smile:

File: scripts/upgrade [hunk 1]
```diff
@@ -64,8 +64,11 @@ path_url=$(ynh_normalize_url_path $path_url)
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
 
+# Create a temporary directory
+tmpdir="$(ynh_smart_mktemp 10)"
```
Comment (kay0u) [inline scripts/upgrade]: In `ynh_clean_setup` you should remove this directory

Comment (JimboJoe) [reply scripts/upgrade]: Hm OK, so we should test that the `$tmpdir` variable is set, and the corresponding directory exists before removing it. Is there an example already implemented somewhere (lazy mode)...? :sleeping:

File: scripts/upgrade [hunk 2]
```diff
@@ -40,6 +40,10 @@ fi
 
 ynh_backup_before_upgrade	# Backup the current version of the app
 ynh_clean_setup () {
+	# Delete any created temporary folder
+	if [ -v tmpdir ] && [ -d $tmpdir ]; then
```
Comment (maniackcrudelis) [inline scripts/upgrade]: What is `-v` ?

Comment (JimboJoe) [reply scripts/upgrade]:
```
-v varname

    True if the shell variable varname is set (has been assigned a value).
```
as explained [here](https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html) (requires Bash 4.2+)

Comment (maniackcrudelis) [reply scripts/upgrade]: Ok, that's the problem between `test` and `[`, the man aren't the same. But this option works for both...
Hopefully it's used after `set +eu` !

File: scripts/upgrade (hunk 1 above)
Comment (JimboJoe) [reply scripts/upgrade]: Implemented in https://github.com/YunoHost-Apps/ttrss_ynh/commit/122ad91d7c727564a1bfb3e46ca18e2c6e13713f
Can you please review it? :wink:

**Comment by kay0u** [comment]
Will be merged in 3 days.

**Pull request merged**